Frends.FTP.UploadFiles/Frends.FTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -0,0 +1,376 @@
+                        case DestinationAction.Overwrite:
+                            PutDestinationFile(removeExisting: true);
+                            break;
+                        case DestinationAction.Error:
+                            throw new DestinationFileExistsException(DestinationFileNameWithMacrosExpanded);
+                    }
+                }
+                else
+                {
+                    PutDestinationFile();
+                }
+
+                if (_batchContext.Options.PreserveLastModified)
+                    RestoreModified();
+
+                ExecuteSourceOperation();
+                _logger.LogTransferSuccess(this, _batchContext);
+                CleanUpFiles();
+            }
+            catch (Exception ex)
+            {
+                var sourceFileRestoreMessage = RestoreSourceFileAfterErrorIfItWasRenamed();
+                HandleTransferError(ex, sourceFileRestoreMessage);
+            }
+            CleanUpFiles();
+            return _result;
+        }
+
+        /// <summary>
+        /// Checks if destination file exists.
+        /// </summary>
+        /// <returns></returns>
+        private bool DestinationFileExists(string filename)
+        {
+            Trace(
+                TransferState.CheckIfDestinationFileExists,
+                "Checking if destination file {0} exists",
+                SourceFile.Name);
+            return _client.FileExists(filename);
+        }
+
+        /// <summary>
+        /// Rename file with unique file name. 
+        /// </summary>
+        private void RenameSourceFile()
+        {
+            var uniqueFileName = Util.CreateUniqueFileName();
+            var workdir = !string.IsNullOrEmpty(_batchContext.Info.WorkDir)
+                ? _batchContext.Info.WorkDir
+                : Path.GetDirectoryName(SourceFile.FullPath);
+            var renamedFile = Path.Combine(workdir, uniqueFileName);
+
+            Trace(TransferState.RenameSourceFileBeforeTransfer, "Renaming source file {0} to temporary file name {1} before transfer", SourceFile.Name, uniqueFileName);
+            File.Move(SourceFile.FullPath, renamedFile);
+
+            _sourceFileDuringTransfer = renamedFile;
+        }
+
+        /// <summary>
+        /// Appends source file content to existing destination file.
+        /// </summary>
+        private void AppendDestinationFile()
+        {
+            Trace(
+                TransferState.AppendToDestinationFile,
+                "Appending file {0} to existing file {1}",
+                SourceFile.Name,
+                DestinationFileNameWithMacrosExpanded);
+
+            // Determine path to use to the destination file.
+            var path = (_batchContext.Destination.Directory.Contains("/")) 
+                ? $"{_batchContext.Destination.Directory}/{DestinationFileNameWithMacrosExpanded}"
+                : Path.Combine(_batchContext.Destination.Directory, DestinationFileNameWithMacrosExpanded);
```

Comment:
There might come up a conflict if destination directory has macros in it. IMO this _batchContext.Destination.Directory has the original input from user so there might be macros which are not handled.